app/Http/Controllers/ProfileController.php

```diff
@@ -178,17 +178,33 @@ public function exportImages()
     {
         abort_if(!\Auth::user()->canAccessFilament(), 403, 'Insufficient permissions');
 
-        $zip = new ZipArchive;
+        $zipName = 'profileImages.zip';
 
-        if (true === ($zip->open('profileImages.zip', ZipArchive::CREATE | ZipArchive::OVERWRITE))) {
-            foreach (Storage::allFiles() as $file) {
-                $name = basename($file);
-                if ($name !== '.gitignore' && $name !== 'profileImages.zip') {
-                    $zip->addFile(Storage::path($name), $name);
+        $zip = new ZipArchive();
+
+        if (true === ($zip->open(Storage::path($zipName), ZipArchive::CREATE | ZipArchive::OVERWRITE))) {
```

Comment:
Won't this potentially cause issues when multiple requests for an export arrive close together, since two backend processes would try to write to the exact same filename and path?

Reply:
Is it required that this file is stored using `Storage` or is there maybe a way we can generate the ZIP in e.g. the system's temporary files storage and delete it again after sending the download to the user?